### DISCUSSION TITLE
关于MyHint的一些事

### DISCUSSION THREAD

**Discussion by Hill23333** [Meloong-Git/PCL#5161] [讨论]
事情源头要从[#5129 ](https://github.com/Hex-Dragon/PCL2/pull/5129)说起了...大家点进去自己看吧
于是我试着做了一下，就是下面这个效果
![image](https://github.com/user-attachments/assets/31c28024-1298-42c5-9e6e-fe63a074b5e9)
（为了兼容性，保留IsWarn，行为不变，Type覆盖IsWarn的行为）

<details>
<summary>但是如果改的话PCL2自身的MyHint就要大改，我找出了以下几个地方：（还不算上PCL2的帮助）</summary>

### PageDownloadInstall.xaml
- 行21 ```<local:MyHint Text="你尚未选择安装 Fabric API，这会导致大多数 Mod 无法使用！" Margin="0,1,0,7" x:Name="HintFabricAPI" IsWarn="True" />```
- 行22 ```<local:MyHint Text="你尚未选择安装 OptiFabric，这会导致 OptiFine 无法使用！" Margin="0,1,0,7" x:Name="HintOptiFabric" IsWarn="True" />```
- 行23 ```<local:MyHint Text="安装结束后，请在 Mod 下载中搜索 OptiFabric Origins 并下载，否则 OptiFine 会无法使用！" Margin="0,1,0,7" x:Name="HintOptiFabricOld" IsWarn="False" />```
- 行24 ```<local:MyHint Text="OptiFine 与一部分 Mod 的兼容性不佳，请谨慎安装。" Margin="0,1,0,7" x:Name="HintModOptiFine" IsWarn="False" />```

###  PageDownloadError.xaml
- 行99 ```<local:MyHint x:Name="HintError" Text="无法连接到 Modrinth，所以目前仅显示了来自 CurseForge 的内容，结果可能不全。&#xa;请尝试使用 VPN 或加速器以改善网络。" IsWarn="True" Margin="0,0,0,15" />```

### PageDownloadPack.xaml
- 行88 ```<local:MyHint x:Name="HintError" Text="无法连接到 Modrinth，所以目前仅显示了来自 CurseForge 的内容，结果可能不全。&#xa;请尝试使用 VPN 或加速器以改善网络。" IsWarn="True" Margin="0,0,0,15" />```

### PageLinkIoi.xaml
- 行13 ```<local:MyHint Margin="5,2,5,8" x:Name="LabHint" IsWarn="False" />```

### PageSetupLaunch.xaml
- 行120 ```<local:MyHint x:Name="LabRamWarn" Text="32 位 Java 最多只能分配 1 GB 内存，建议安装 64 位 Java!" IsWarn="True" Margin="0,0,0,12" />```
- 行175 ```<local:MyHint Text="由于技术问题，此功能只保证对 1.19.2 以前的版本有效！" IsWarn="True" Margin="0,0,0,10" />```

### PageSetupUI.xaml
- 行209 ```<local:MyHint Margin="0,15,0,1" Text="内容见 ModSetup" IsWarn="False" x:Name="HintCustom" />```

### PageVersionSetup.xaml
- 行10 ```<local:MyHint Text="本页面的设置均只对当前版本生效，这对整合包制作应该会有所帮助。&#xa;你也可以在设置页面中修改对全部版本生效的全局设置。" IsWarn="False" Margin="0,15" />```
- 行61 ```<local:MyHint x:Name="LabRamWarn" Text="32 位 Java 最多只能分配 1 GB 内存，建议安装 64 位 Java!" IsWarn="True" Margin="0,0,0,12" />```

~还真不少...~

</details>

总而言之，我希望大家可以一起讨论一下这些地方的IsWarn替换成什么更好。是Note?Warning?Caution?这件事我自己一个人做不了主。

@JingHai-Lingyun

**Comment by MoYuan-CN** [comment]
黄色…… 对比度…… 哇呀……
试试看 bg `#f9e7c3` fg `#e69900` 的效果？

**Reply by Hill23333** [reply]
![image](https://github.com/user-attachments/assets/de54637e-dd05-4842-a949-766b6ce2ecf9)
是不是好一点？

**Reply by MoYuan-CN** [reply]
稍微好看一点了，但是对比度还是不太好……
fg `#573a00` 试试看

**Reply by Hill23333** [reply]
![image](https://github.com/user-attachments/assets/072cff04-49fd-4c02-8309-8f545191be55)
感觉有一点奇怪...

**Reply by MoYuan-CN** [reply]
调颜色是最难受的事情……
来社区群聊吧，在这里聊容易炸邮箱

**Reply by Hill23333** [reply]
![image](https://github.com/user-attachments/assets/6bcbb12a-5c68-4260-aa14-97c773ddc388)
试了一下```#86861d```，是不是还好？

**Reply by Hill23333** [reply]
> 调颜色是最难受的事情…… 来社区群聊吧，在这里聊容易炸邮箱

好

**Comment by Hill23333** [comment]
调色的问题解决了。另一个问题是是否要将目前PCL内使用MyHint的地方的IsWarn全部换成Type？换成哪种提示比较好？（非常感谢PCL Community的人帮忙调色！）还是说先提交一个新功能issue让龙猫决定比较好...

**Reply by LinQingYuu** [reply]
其实开个 PR 就行(((

**Reply by Hill23333** [reply]
直接开pr就行吗？不需要提前交个新功能issue吗？我不太清楚...

**Reply by Pigeon0v0** [reply]
> 直接开pr就行吗？不需要提前交个新功能issue吗？我不太清楚...

直接开就可以

**Reply by Hill23333** [reply]
> > 直接开pr就行吗？不需要提前交个新功能issue吗？我不太清楚...
> 
> 直接开就可以

好

**Reply by LinQingYuu** [reply]
>直接开pr就行吗

因为你开个 Issue 还得花点精力查重（
PR 少，所以无所顾虑（除非也重了（（（

**Comment by Hill23333** [comment]
最后终于开了 pr，谢谢各位！

**Reply by MoYuan-CN** [reply]
后续讨论转
- #5177
